### PR TITLE
Disable errors via inline comments

### PR DESCRIPTION
**Pull request by pantor**
This PR adds the possibility to disable specific errors at specific lines via an inline comment like
```vala
if(...) { // vala-lint=space-before-paren, line-length
```
This is common behaviour for a linter. That way, all code style errors in vala-lint can be fixed by disabling the (needed) warnings in the `UnitTest.vala` file.

### CODE REVIEW

File: lib/DisableResult.vala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+public struct ValaLint.DisableResult {
+    public string check_title;
+    public int line_pos;
+}
```
Comment (jeremypw) [inline lib/DisableResult.vala]: Is a separate file needed for a single struct?

File: lib/Disabler.vala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+public class ValaLint.Disabler : Object {
+    const string DISABLE_KEY = "vala-lint";
```
Comment (jeremypw) [inline lib/Disabler.vala]: Maybe "vala-lint-ignore" would be clearer.

File: lib/Disabler.vala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     * Parses a list of checks to disable at specific lines.
+     *
+     * @param parse_result The parse result from ValaLint.Parser.
+     */
+    public Vala.ArrayList<DisableResult?> parse (Vala.ArrayList<ParseResult?> parse_result) {
+        var result = new Vala.ArrayList<DisableResult?> ();
+
+        foreach (ParseResult r in parse_result) {
+            if (r.detail_type == ParseDetailType.INLINE_COMMENT) {
+                int index = r.text.index_of (@"$DISABLE_KEY=");
```
Comment (jeremypw) [inline lib/Disabler.vala]: Could include "=" in key? So could use index_of (DISABLE_KEY).

Comment (pantor) [reply lib/Disabler.vala]: '=' is now included in the key. I would prefer a clear syntax with either '=' or ':', at the moment I've just followed pylint using '='.

File: lib/DisableResult.vala (hunk 1 above)
Comment (pantor) [reply lib/DisableResult.vala]: Included the struct in the disabler file.

**Pull request merged**